DevWs for HAB Home Intelligence containing containing Z-Wave Window Shade

habhomegit opened the pull request.

iblinds committed:
DevWs for HAB Home Intelligence containing containing Z-Wave Window Shade

shrakuma commented:
@habhomegit would you be able to send the devices to our certification center in Poland. We can communicate the info to you via email on the next steps. The reason is becuase, as this is a custom DTH, we have to review that with the device in hand. In some cases, we may have a better option of including the FIngerprint of this device into one of our generic DTHs (if a generic DTH fits the capabilities of the devoce)
hence please reach out to me at shravan.kumar@smartthings.com and we shall provide the shipment info. There is no other cost associated with our review, testing and publishing the DTH.

shrakuma commented:
@KKlimczukS @PKacprowiczS  can you please take a look at this DTH . it's a custom DTH but it would be helpful if you guys can review this for checking only for  the areas that may be concerning for our platform. Essentially we don't need to check the functionality line by line as this is a CST request. 
We have received the confirmation from the partner that the DTH is working as expected

KKlimczukS commented on devicetypes/iblinds/iblinds-zwave.src/iblinds-zwave.groovy:
```diff
@@ -0,0 +1,299 @@
+        capability "Window Shade" 
+        capability "Window Shade Preset"	
+
+        command "stop"
```
Custom commands are not supported in the SmartThings application (they are still supported in the SmartThings Classic only). Although the UI (generic-shade-3) handles pause().

habhomegit replied:
command "Stop" is in your published Z-Wave-Window-Shade DTH

KKlimczukS commented on devicetypes/iblinds/iblinds-zwave.src/iblinds-zwave.groovy:
```diff
@@ -0,0 +1,299 @@
+        command "stop"
+
+
+        fingerprint mfr:"0287", prod:"0003", model:"000D", deviceJoinName: "iBlinds Motor"
```
According to new fingerprint writing rules - deviceJoinName should be changed to "iBlinds Window Treatment"

habhomegit replied:
Updated in lastest commit

KKlimczukS commented on devicetypes/iblinds/iblinds-zwave.src/iblinds-zwave.groovy:
```diff
@@ -0,0 +1,299 @@
+        reply "200163,delay 1000,2602": "command: 2603, payload: 10 63 FE"
+    }
+
+    tiles(scale: 2) {
```
Whole "tiles" section will be handled in the "SmartThings Classic" only.

habhomegit replied:
Correct,  This is from the  published Z-Wave-Window-Shade DTH

KKlimczukS commented on devicetypes/iblinds/iblinds-zwave.src/iblinds-zwave.groovy:
```diff
@@ -0,0 +1,299 @@
+import groovy.json.JsonOutput
+
+
+metadata {
```
Spacing (in the whole file) should be changed (4 spaces -> 1 tab).

habhomegit replied:
Will fix

habhomegit replied:
Updated

KKlimczukS commented on devicetypes/iblinds/iblinds-zwave.src/iblinds-zwave.groovy:
```diff
@@ -0,0 +1,299 @@
+        fingerprint mfr:"0287", prod:"0004", model:"0071", deviceJoinName: "iBlinds Motor"
+    }
+
+    simulator {
```
"simulator" section is not required.

habhomegit replied:
Again, I left this in because the Simulator section is in your published Z-Wave-Window-Shade DTH

KKlimczukS commented on devicetypes/iblinds/iblinds-zwave.src/iblinds-zwave.groovy:
```diff
@@ -0,0 +1,299 @@
+}
+
+def updated() {
+    if (device.latestValue("checkInterval") != checkInterval) {
```
It may not be needed here (since checkInterval has been set in installed() already).

habhomegit replied:
From  published Z-Wave-Window-Shade DTH

KKlimczukS commented:
@shrakuma, since the code is based (there a minor changes here) on the existing DTH: "Z-Wave Window Shade" - Could this device be handled there ?

habhomegit commented:
@KKlimczukS @shrakuma 

Yes I started with DTH Z-Wave-Window Shade and made a few tweaks.  

•	Added Switch and Switch Level for better Amazon Echo Support (works best when these come before Window Shade Capabilities)
•	Added “Reverse” Preference for users that want their blinds to tilt in the opposite direction
•	Check for Reverse In Close and Set Level functions
•	Updated Fingerprint

iblinds committed:
Updated spaces to Tab  
updated iblinds motor to "IBlinds Window Treatment"

shrakuma commented:
@dkirker @KKlimczukS 
@habhomegit based on our internal discussions with our hub team , the recomended approach is to submit a PR against Z-Wave Window Shade to add the support for “reverse”. we are handling the alexa integration in a different way. one of our team members can add additional info on the topic

habhomegit commented:
@dkirker @KKlimczukS @shrakuma, 

Sorry PR = Pull Request 👍  I didn't catch the acronym. PR in my daily world means Problem Report. 

I'd like to "test" and include the Alexa Support in the commit as well as my reverse change.   Can you please point me to documentation for this or explain it to me. 

Best regards, 
Eric

habhomegit commented:
I submitted a request through the ST IDE but I'm not sure this was the correct procedure.   Should I delete my request in the Developer Workspace and start a new one?  Sorry I'm a little confused

dkirker committed:
Prepare iblinds-zwave for deploy

dkirker committed:
Update deviceJoinName to match new naming pattern

dkirker commented:
I updated this PR to get it ready so that it can be deployed as CST (I have spoken with @shrakuma and this is ok). We'll start the device as CST and then continue to make improvements in https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/29999 to further enhance the PR. That work will then get the device to WWST status.

habhomegit commented:
Thanks for the updates @dkirker

dkirker commented:
@habhomegit Sure thing! I just sent an email with some more details.
